scenario-compiler/src/main/scala/pl/touk/nussknacker/engine/extension/CastOrToConversionExt.scala

```diff
@@ -0,0 +1,165 @@
+package pl.touk.nussknacker.engine.extension
+
+import cats.data.ValidatedNel
+import cats.implicits.catsSyntaxValidatedId
+import pl.touk.nussknacker.engine.api.generics.{GenericFunctionTypingError, MethodTypeInfo, Parameter}
+import pl.touk.nussknacker.engine.api.typed.typing
+import pl.touk.nussknacker.engine.api.typed.typing.{Typed, TypedObjectWithValue, TypingResult, Unknown}
+import pl.touk.nussknacker.engine.definition.clazz.{ClassDefinitionSet, FunctionalMethodDefinition, MethodDefinition}
+import pl.touk.nussknacker.engine.extension.CastOrToConversionExt.getConversion
+import pl.touk.nussknacker.engine.util.Implicits.RichScalaMap
+import pl.touk.nussknacker.engine.util.classes.Extensions.{ClassExtensions, ClassesExtensions}
+
+import java.lang.{Boolean => JBoolean}
+import scala.util.Try
+
+// todo: lbg - add casting methods to UTIL
+class CastOrToConversionExt(target: Any, classesBySimpleName: Map[String, Class[_]]) {
```

Comment:
1. This `To` is redundant here or I missing some concept?
2. We have to handle implicit conversions here as well. See `TypeConversionHandler` for details. For example, I assume that currently `1.to('Long')` will work but `1.to('Float')` won't work. Am I right? We also have implicit String -> some date types conversions - see `TypeConversionHandler.stringConversions`. We can add them to the chain in a separate PR if you want

Reply:
1. Done
2. Next PR is a great idea

Reply:
1. There are still some places not changed, search for `castOrTo`

Reply:
Done